synapse/rest/client/room.py

```diff
@@ -389,14 +408,13 @@ def register(self, http_server: HttpServer) -> None:
         PATTERNS = "/join/(?P<room_identifier>[^/]*)"
         register_txn_path(self, PATTERNS, http_server)
 
-    async def on_POST(
+    async def _do(
```

Comment:
(Random place for a comment.)

I don't expect any changes from this, but wow, this ends up being very repetitive! It makes me wonder if `TransactionRestServlet` should define an abstract `_do` (or something else) method and provide the `on_PUT` and `on_POST` methods automatically. That seems like a bigger change than we need here, but wanted to leave a comment!

Reply:
Yeah, I also thought of that. But making typing work for that, with potentially very different arguments discouraged me a bit from trying 🫣